### PR TITLE
feat: dump `run/${type}_config_meta.json`

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Show who define the property of the config

Closes https://github.com/eggjs/egg/issues/1132

### CODE REVIEW

**Comment by popomore** [comment]
依赖 https://github.com/eggjs/egg-core/pull/106

**Comment by fengmk2** [comment]
https://github.com/eggjs/egg/pull/1152 看来等你这个功能一起发吧，发 minor

**Comment by fengmk2** [comment]
https://github.com/eggjs/egg-core/pull/107 已经修复

**Comment by popomore** [comment]
再看看还有啥改的么

**Comment by fengmk2** [comment]
没了，合并。

File: test/lib/agent.test.js [hunk 1]
```diff
@@ -21,16 +21,16 @@ describe('test/lib/agent.test.js', () => {
...
-          done();
-        }, 1000);
-      });
+        .get('/agent-throw')
+        .expect(200, err => {
+          assert(!err);
+          setTimeout(() => {
+            const body = fs.readFileSync(path.join(baseDir, 'logs/agent-throw/common-error.log'), 'utf8');
+            assert(body.includes('nodejs.unhandledExceptionError: agent error'));
+            app.notExpect(/nodejs.AgentWorkerDiedError/);
```
Comment (atian25) [inline test/lib/agent.test.js]: it's a bug, should be `app.notExpect('stderr', /nodejs.AgentWorkerDiedError/);`

coffee@4 will ignore unknown type, by coffee@4 will throw and help us to find wrong test fixture like this.

**Pull request merged**